torch/csrc/cuda/Stream.cpp

```diff
@@ -38,27 +37,46 @@ static PyObject * THCPStream_pynew(PyTypeObject *type, PyObject *args, PyObject
 
   THCPStream* self = (THCPStream *)ptr.get();
   self->cdata = stream.pack();
-  self->device = stream.device_index();
-  self->cuda_stream = stream.stream();
+  self->cuda_stream = stream;
   return (PyObject *)ptr.release();
   END_HANDLE_TH_ERRORS
 }
 
 static PyObject * THCPStream_query(THCPStream *self) {
   HANDLE_TH_ERRORS
-  return PyBool_FromLong(at::cuda::CUDAStream::unpack(self->cdata).query());
+  return PyBool_FromLong(self->cuda_stream.query());
+  END_HANDLE_TH_ERRORS
+}
+
+static PyObject * THCPStream_device(THCPStream *self) {
+  HANDLE_TH_ERRORS
+  return PyLong_FromLong(self->cuda_stream.device_index());
```

Comment:
Prefer `THPUtils_packInt64` from `python_numbers.h`. It has slightly nicer behavior in Python 2 (preferring `int` when the value is small enough) .